conf/settings.py

```
@@ -361,21 +361,29 @@
+AWS_S3_DATA_SCIENCE_BINDING_BUCKET_NAME = env.str('AWS_S3_DATASCIENCE_BINDING_BUCKET_NAME')
+AWS_STORAGE_BUCKET_NAME_DATA_SCIENCE = ''
+
 if 'aws-s3-bucket' in VCAP_SERVICES:
-    credentials = VCAP_SERVICES['aws-s3-bucket'][0]['credentials']
-    AWS_ACCESS_KEY_ID = credentials['aws_access_key_id']
-    AWS_SECRET_ACCESS_KEY = credentials['aws_secret_access_key']
-    AWS_STORAGE_BUCKET_NAME = credentials['bucket_name']
-    AWS_S3_REGION_NAME = credentials['aws_region']
-    AWS_S3_ENCRYPTION = True
-    AWS_DEFAULT_ACL = None
-else:
-    AWS_S3_ENCRYPTION = False
-    AWS_ACCESS_KEY_ID = env.str('AWS_ACCESS_KEY_ID', '')
-    AWS_SECRET_ACCESS_KEY = env.str('AWS_SECRET_ACCESS_KEY', '')
-    AWS_STORAGE_BUCKET_NAME = env.str('AWS_STORAGE_BUCKET_NAME', '')
-    AWS_S3_REGION_NAME = env.str('AWS_S3_REGION_NAME', '')
-    AWS_DEFAULT_ACL = 'public-read'
+    for aws_paas_bucket in VCAP_SERVICES['aws-s3-bucket']:
+        if aws_paas_bucket['name'] == AWS_S3_DEFAULT_BINDING_BUCKET_NAME:
+            credentials = aws_paas_bucket['credentials']
+            AWS_ACCESS_KEY_ID = credentials['aws_access_key_id']
+            AWS_SECRET_ACCESS_KEY = credentials['aws_secret_access_key']
+            AWS_STORAGE_BUCKET_NAME = credentials['bucket_name']
+            AWS_S3_REGION_NAME = credentials['aws_region']
+            AWS_S3_ENCRYPTION = True
+            AWS_DEFAULT_ACL = None
+        elif aws_paas_bucket['name'] == AWS_S3_DATA_SCIENCE_BINDING_BUCKET_NAME:
+            credentials = aws_paas_bucket['credentials']
+            AWS_ACCESS_KEY_ID_DATA_SCIENCE = credentials['aws_access_key_id']
+            AWS_SECRET_ACCESS_KEY_DATA_SCIENCE = credentials['aws_secret_access_key']
+            AWS_STORAGE_BUCKET_NAME_DATA_SCIENCE = credentials['bucket_name']
+            AWS_S3_REGION_NAME_DATA_SCIENCE = credentials['aws_region']
+            AWS_S3_ENCRYPTION_DATA_SCIENCE = True
+            AWS_DEFAULT_ACL_DATA_SCIENCE = None
```

the looping approach will hide an error if we do a typo in `AWS_S3_DEFAULT_BINDING_BUCKET_NAME` or `AWS_S3_DATA_SCIENCE_BINDING_BUCKET_NAME`:

the blocks that sets `AWS_ACCESS_KEY_ID` will never be hit if `aws_paas_bucket['name'] == AWS_S3_DEFAULT_BINDING_BUCKET_NAME` is not true. That can happen if we doa  typo in vault

this will result in error later when code tries to use those settings

we can make the error happen sooner at app start up by doing something like this:

```
bucket_credentials_map = {
    item['name']: item
    for item in VCAP_SERVICES['aws-s3-bucket']
}
default_credentials = bucket_credentials_map[AWS_S3_DEFAULT_BINDING_BUCKET_NAME]
datascience_credentials = bucket_credentials_map[AWS_S3_DATA_SCIENCE_BINDING_BUCKET_NAME]

```